### PR TITLE
Update angular-resource.d.ts

### PR DESCRIPTION
**Pull request by ilgb**
Replace methods return type T by IResource<T> in IResourceClass<T>

### CODE REVIEW

**Comment by vvakame** [comment]
@diegovilar @miffels could you review this PR?

**Comment by miffels** [comment]
Seems correct, no idea how I missed it back then: [Source](https://github.com/angular/angular.js/blob/v1.3.16/src/ngResource/resource.js#L556)

:+1:

**Comment by sandcastle** [comment]
@miffels what are you looking as far as tests? The existing test file contains calls for all of the relevant methods:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/angularjs/angular-resource-tests.ts#L38-L84

/CC @ilgb

**Comment by miffels** [comment]
@sandcastle Oh, I did not assume the change was backwards-compatible with the tests, so I expected the tests had to be changed, to. Turns out this addition actually is covered.

:shipit: :+1:

File: angularjs/angular-resource.d.ts [hunk 1]
```diff
@@ -71,12 +71,12 @@ declare module angular.resource {
...
     // https://github.com/angular/angular.js/blob/v1.2.0/src/ngResource/resource.js#L538-L549
     interface IResourceClass<T> {
-        new(dataOrParams? : any) : T;
-        get(): T;
-        get(params: Object): T;
-        get(success: Function, error?: Function): T;
-        get(params: Object, success: Function, error?: Function): T;
-        get(params: Object, data: Object, success?: Function, error?: Function): T;
+        new(dataOrParams? : any) : IResource<T>;
+        get(): IResource<T>;
```
Comment (vvakame) [inline angularjs/angular-resource.d.ts]: um? is this right?
now typescript is support intersection types. https://github.com/Microsoft/TypeScript/pull/3622
I think we should use `get(): T & IResource<T>;`. isn't it?

**Comment by RyanCavanaugh** [comment]
Hello and thank you for your contribution!

Due to an excessively long queue of pull requests that have become stale, we are "declaring bankruptcy" and closing all PRs opened before May 1, 2016. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

Going forward, we are committing to review or merge all PRs on a regular basis so this bankruptcy will not occur again. We apologize for the incovenience and hope you will continue to contribute to DefinitelyTyped in the future.

Thanks
The TypeScript and DefinitelyTyped teams

**Pull request closed without merging**